Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -99,16 +101,23 @@ class DatabaseContainer: NSPersistentContainer {
         deletedMessagesVisibility: ChatClientConfig.DeletedMessageVisibility? = nil,
         shouldShowShadowedMessages: Bool? = nil
     ) {
-        // It's safe to unwrap the following values because this is not settable by users and it's always a programmer error.
-        let bundle = bundle ?? Bundle(for: DatabaseContainer.self)
-        let modelURL = bundle.url(forResource: modelName, withExtension: "momd")!
-        let model = NSManagedObjectModel(contentsOf: modelURL)!
+        let managedObjectModel: NSManagedObjectModel
+        if let cachedModel = Self.cachedModel {
+            managedObjectModel = cachedModel
+        } else {
+            // It's safe to unwrap the following values because this is not settable by users and it's always a programmer error.
+            let bundle = bundle ?? Bundle(for: DatabaseContainer.self)
+            let modelURL = bundle.url(forResource: modelName, withExtension: "momd")!
+            let model = NSManagedObjectModel(contentsOf: modelURL)!
+            managedObjectModel = model
+        }
+        Self.cachedModel = managedObjectModel
```

Comment:
this line should go in the `else`.

Reply:
Right now, changing this does not impact the logic. Do you think it is more readable, tho?

Reply:
Changed it so that the logic is easier to understand 👍